pyproject.toml

```diff
@@ -1,2 +1,5 @@
 [build-system]
-requires = ["setuptools", "cython"]
+requires = ["setuptools<72", "cython>=3.1"]
```

Comment:
This change can be dropped, neither pin is necessary.

Reply:
Not sure about dropping the `setuptools` pin, see https://github.com/Quansight-Labs/ndindex/pull/199#discussion_r2082373269

and also fails consistently for pp39 and macos here: https://github.com/Quansight-Labs/ndindex/actions/runs/15003261213/job/42155535707?pr=199